Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamer.cpp

```diff
@@ -1501,14 +1501,15 @@ void MediaPlayerPrivateGStreamer::updateTracks(const GRefPtr<GstObject>& collect
         return;
 
     using TextTrackPrivateGStreamer = InbandTextTrackPrivateGStreamer;
-#define CREATE_OR_SELECT_TRACK(type, Type) G_STMT_START { \
+#define CREATE_OR_SELECT_TRACK(type, Type, caps) G_STMT_START { \
         if (!m_##type##Tracks.contains(streamId)) { \
             auto track = Type##TrackPrivateGStreamer::create(*this, type##TrackIndex, stream); \
             if (player)                                                 \
                 player->add##Type##Track(track);                        \
             m_##type##Tracks.add(streamId, WTFMove(track));             \
         }                                                               \
         auto track = m_##type##Tracks.get(streamId);                    \
+        track->updateConfigurationFromCaps(WTFMove(caps));              \
```

Comment:
This makes me a bit uneasy because it's a macro but in the end it resembles a function. I think you should move in the caller and avoid moving here.

Reply:
I don't think that's possible, the macro is not a function.